src/crawlers/basic_crawler.js

```diff
@@ -262,7 +262,13 @@ export class BasicCrawler {
         this.userProvidedHandler = handleRequestFunction;
         this.failedContextHandler = handleFailedRequestFunction;
         this.handleRequestTimeoutMillis = handleRequestTimeoutSecs * 1000;
-        this.internalTimeoutMillis = Math.max(this.handleRequestTimeoutMillis, 300e3); // allow at least 5min for internal timeouts
+        const tryEnv = (val) => (val == null ? val : +val);
+        // allow at least 5min for internal timeouts
+        this.internalTimeoutMillis = tryEnv(process.env.APIFY_INTERNAL_TIMEOUT) ?? Math.max(this.handleRequestTimeoutMillis, 300e3);
```

Comment:
I would go with something like `this.handleRequestTimeoutMillis * 2 + 10`. There are all the internal SDK function calls that need to complete as well.

Reply:
The timeout was already there, all I changed was to make it configurable via env vars. It's been in production for about 4 months, so I'd say the value is fine. But we can double it for the new RQ reset check to be reallu sure.

Note that this value is used as a timeout for several things separately, it is not meant to be some "master timeout that fits all for a request life cycle", it is used for the internal API calls, and also as [a timeout for the user handler](https://github.com/apify/apify-js/blob/master/src/crawlers/basic_crawler.js#L483-L489).

Reply:
Even if it was there, that does not change the validity of my argument that such a timeout has a race condition 😄